File: core-java-practice/leet-code-codebase/addDigits.java
import java.util.*;
public class addDigits {
    public static void main(String[] args) {
        Scanner s = new Scanner(System.in);
        int num = s.nextInt();
        while (num >= 10) { //jab tak single digit hai tab tak chalega
            int sum = 0;
            while (num > 0) {
                sum += num % 10; //har digit ko sum mai add karenge
                num /= 10;
            }
            num = sum;
        }
        System.out.println(num);
        s.close();
    }
}

File: core-csharp-practice/leet-code-practice/AddDigits.cs
using System;

class AddDigits
{
    static void Main()
    {
        int num = 9875;   // fixed number

        while (num >= 10)
        {
            int sum = 0;

            while (num > 0)
            {
                sum += num % 10;
                num /= 10;
            }

            num = sum;
        }

        Console.WriteLine(num);
    }
}
